### DISCUSSION TITLE
Grafana query/dashboard to show and search logs?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [serilog-contrib/serilog-sinks-grafana-loki#229] [Q&A] [answered]
Hello, I am new to Grafana. I have my app sending logs to Loki, and i have my dashboard showing said logs:

![image](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/assets/2934507/ef053318-0f47-4847-9796-1cdb591c44ac)

My dashboard is currently very ugly. Can someone share a way to show maybe only the Message, level, time and allow search?

**Comment by mishamyte** [comment, marked as answer]
Hmmm, never used dashboards for that.

If you wanna work with logs as with logs, usually you go "Explore -> Loki", then use few labels for filtering, adding a JSON parser into the pipeline and work with the structured logs then.

![image](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/assets/5649656/2ed1543c-8fea-46f8-afe3-ca67367942df)

**Reply by VictorioBerra** [reply]
Thank you this is helpful! I am self-hosting some .NET apps and I was looking for a free Splunk-like experience. There is not many options for centralized logging for free. Seq is good but its free version has limited features.

**Reply by mishamyte** [reply]
Yeah, I understand. 

TBH I worked over different solutions in last years: ELK, Loki, Seq, Datadog etc. 

I like Seq for logs, but as you mentioned, it has some limitations on a free tier. Plus often it could be out of the system context (when you using complex APM solutions). 

Loki is on my 2nd place, it has amazing integration with Tempo (so you have logs + traces combined). And works perfectly with OTEL. 
But it requires some knowledge how to cook it for the structured logging (and not shoot yourself in the leg).

BTW, probably, you have read it already, but I would suggest to check the next article in that project's Wiki: [Using LokiJsonTextFormatter and Grafana Loki json parser
](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/wiki/Using-LokiJsonTextFormatter-and-Grafana-Loki-json-parser)

It would help you to avoid some common issues (like a high cardinality)
